### ISSUE TITLE
Sections between two Horizontal Rules are not checked

### ISSUE THREAD

**Issue by jonathanbaker7** [lukeapage/node-markdown-spellcheck#98]
In a markdown file that contains 2 or more Horizontal Rules, using the triple-dash syntax (---), the spell checking system will not check anything inside a matched set of dashes.

**Reproduction:**

The following markdown file is legal.  It has a header, which is not spell checked.  It also has a legal section, defined by two horizontal rule markings, which is also not checked.

```markdown
---
Legal Header.  The spell checking system removes this and does not check it.
---

Some valid text...

---

In a prexxious file, the test slervice called spellering was set up for all applicaitons.  Next, this specific applicaiton must connectetctecxxtaasrs to that "Destination".

Misspellings can occur inside this section, because it is viewed as a header.  They will not be caught.

---
```

**Suspected Cause:**

The problem appears to be in `markdown-parser.js` line 13.  This line removes all of the sections between instances of horizontal rules (---):

```javascript
tracker.replaceAll(/\---\r?\n[\w\W]*?\r?\n---/, " "); // remove header
```

**Comment by truist** [comment]
Per [this StackOverflow post](https://stackoverflow.com/questions/44215896/markdown-metadata-format), it appears that node-markdown is trying to emulate a Jekyll feature that allows for a metadata section at the top of the markdown file, by ignoring it. But the consequence (that it then ignores everything between two HRs) seems worse than just not having the feature.

**Comment by jonathanbaker7** [comment]
I agree, it would be useful to have that feature.  But, as that feature uses the exact same delimiter as a Horizontal Rule, there should be a switch to turn it on/off.  

And, as you said, the real bug is that it removes all instances of these pairs, not just the first one.

**Comment by jlutz777** [comment]
I agree that there should either be a way to toggle this, use a different delimiter, or remove it. We were missing spelling mistakes due to this and didn't know it.

**Comment by lukeapage** [comment]
Fixed in 1.2.0
